### PR TITLE
Merge in upstream changes to i3lock

### PR DESCRIPTION
**Pull request by mortie**
I haven't tested everything here extensively, but reading through the code, it all seems correct enough. Basic functionality works, and the example script still does what it's supposed to. A lot of the new code is for supporting reading raw images, which I added in https://github.com/i3/i3lock/pull/226.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,27 @@ A [sample script](https://github.com/PandorasFox/i3lock-color/blob/master/lock.s
 On OpenBSD the `i3lock` binary needs to be setgid `auth` to call the
 authentication helpers, e.g. `/usr/libexec/auth/login_passwd`.
 
+Building i3lock
+---------------
+We recommend you use the provided package from your distribution. Do not build
```
Comment (PandorasFox) [inline README.md]: Remove this paragraph, as i3lock-color doesn't generally have packages available from distros.

File: i3lock.c [hunk 2]
```diff
@@ -1829,9 +1992,12 @@ int main(int argc, char *argv[]) {
             case 999:
                 debug_mode = true;
                 break;
+            case 998:
+                image_raw_format = strdup(optarg);
+                break;
             default:
                 errx(EXIT_FAILURE, "Syntax: i3lock [-v] [-n] [-b] [-d] [-c color] [-u] [-p win|default]"
-                                   " [-i image.png] [-t] [-e] [-f]\n"
+                                   " [-i image.png] [-t] [-e] [-f] [-l]\n"
```
Comment (PandorasFox) [inline i3lock.c:2007]: I don't see the case for `l` - I think that the tty lock switching isn't being checked.

**Review by PandorasFox — CHANGES_REQUESTED** [review]
Generally looks fine; couple of minor touches and it should be fine. Thanks for the large merge!

**Comment by mortie** [comment]
Those two things should be fixed now.

**Comment by PandorasFox** [comment]
Thanks! I haven't had the time to sit down and do this myself; big thanks for helping keep this from falling behind :)

**Comment by mortie** [comment]
No problem :)

One thing though; `load_image` takes two arguments now, the `image_path` and `image_raw_format`; when you resolved the conflict with my other PR, `load_image` ended up being called with only one argument.

I'm sorry that I ended up submitting two conflicting PRs, I don't know if there'd be a better way to do it. I felt that it was cleaner to have the two things (merging in upstream as faithfully as possible, change an i3lock-color-specific behavior) as separate PRs, but maybe that wasn't the best idea considering they touch some of the same code.

**Comment by PandorasFox** [comment]
Two merges was totally alright! The merge issue is on me for thinking I could handle a merge conflict from my phone - I should have looked a bit more closely.

**Pull request merged**